lib/library.rb

```diff
@@ -0,0 +1,76 @@
+require 'yaml'
+require 'date'
+require 'pry-byebug'
+
+class Library
+  attr_reader :collection, :return_date
+
+  def initialize
+    @collection = load_book_collection
+  end
+
+  def book_check_out(item, patron)
+    book = find_book(item)
+    case
+    when book == 'no book'
+      'Library does not have this book'
+    when book[:available] == false
+      'Book is not available. It will be returned ' +
+        (@collection[book[:index]][:return_date]).to_s
+    else
+      perform_transaction(book, patron)
+    end
+  end
+
+  def available_books
+    avail_books = []
+    @collection.each do |book|
+      avail_books << book if book[:available] == true
+    end
+    avail_books
+  end
+
+
+  private
+
+  def load_book_collection
+    YAML.load_file('./lib/data.yml')
+  end
+
+  def update_book_collection
+    File.open('./lib/data.yml', 'w') { |f| f.write collection.to_yaml }
+  end
+
+  def perform_transaction(book, patron)
+    # I assume each item has will be unique in the data.yml
+    # (would have liked an book ID in data.yml)
+    i = book[:index]
+    set_book_as_borrowed(i)
+    patron.borrowed_books << @collection[i]
+    update_book_collection
+    @collection[i]  # maybe not so nice?
+        #Line added so method will return a value to check against test.
+  end
+
+  def set_return_date
+    Date.today + 30
```

Comment:
There's a method on the Date class that will give you the date a month from now 😉